sbi/inference/posteriors/likelihood_based_posterior.py

```diff
@@ -92,12 +96,18 @@ def log_prob(
         warn(
             "The log probability from SNL is only correct up to a normalizing constant."
         )
-
         with torch.set_grad_enabled(track_gradients):
+            log_likelihood_across_trials = torch.zeros(
+                theta.shape[0], device=self._device
+            )
+            theta_on_device = theta.to(self._device)
+            # Pass repeated x for every trial, take log prob sum across trials.
+            for x_trial in x.to(self._device):
```

Comment:
right, I actually just implemented it that because I realised the speed-down in MCMC when having 1000 trials. for 1000 trials in could become problematic though when the `theta` batch is large as well, no?